core/app/models/spree/stock/simple_coordinator.rb

```diff
@@ -85,15 +87,8 @@ def split_packages(initial_packages)
       end
 
       def allocate_inventory(availability_by_location)
-        availability_by_location.transform_values do |available|
-          # Find the desired inventory which is available at this location
-          packaged = available & @desired
-
-          # Remove found inventory from desired
-          @desired -= packaged
-
-          packaged
-        end
+        Spree::Deprecation.warn 'allocate_inventory is deprecated. Please write your own allocator defining' \
+          'a Spree::Stock::Allocator::Base subclass'
```

Comment:
This breaks backward compatibility. It's just telling the user to don't use this method but it's also removing what the method does without notice. We should actually call something here, trying to emulate exactly what it was doing but using the new stuff added

Reply:
Unfortunately, I can't use the new stock allocation system for the backward compatibility for two reasons:

1. The `allocate_on_hand`, `allocate_backorderable` and `allocate` method of `availability` allocator are private methods.
2. The desired stock quantities are subtracted into`allocate_inventory` method

Have you any ideas?